Feature/readme game instructions

euqueme opened the pull request:
We work on  the following aspects:

- Improve the game: Now it's possible to have multiple games with the same players.
- The Readme file not only includes the rules for the game and how to run it from a computer also has a live version that will run on repl.it once this project is merged to the master branch.

Thank you in advance for your kind review.

SantiagoGuerra committed:
GAME.RB FILE

SantiagoGuerra committed:
BOARD.RB FILE

SantiagoGuerra committed:
PLAYER.RB FILE

SantiagoGuerra committed:
PLAYER CLASS
create player class to store player information

euqueme committed:
BOARD AND GAME

Create winning combinations and running method in the Game class

Create methods to change the cell and render the board in the Board class

euqueme committed:
Update game.rb

SantiagoGuerra committed:
BOARD CLASS

SantiagoGuerra committed:
GAME CLASS
create detect_winner method, this method will detect a winner and will return his/her name.

euqueme committed:
BOARD AND GAME

Make the Board class independent by not loading the player in the change_cell method

Create initialize, play, next_to_play and not_a_tie methods to switch between players and also validate if the game is a tie in the Game class

optimize  detect winner  and modify running methods to test the Game class.

The code passes all rubocops.

SantiagoGuerra committed:
GAME CLASS
change load

SantiagoGuerra committed:
MAIN FILE
add basic user interface

SantiagoGuerra committed:
UTILS FILE
create utils file to store methods

SantiagoGuerra committed:
MAIN.RB FILE
add interface to user and improve validation system.

euqueme committed:
MULTI GAME

Improve the game by adding multi game functionality so the players won´t need to write their names every time they want to play.

euqueme committed:
README AND RUBOCOP

Add descriptive Readme file including the rules of the game and a live version that will be available once this project is merged to the master branch

Also fix rubocop offenses in the main.rb file

euqueme committed:
COMMENT IDENTATION

Fix them.

stratospherique reviewed (changes requested):
### Great work on Making such a good readme file :clap: However, I was surprised that the game is not working as it should. Would you check the case below:
```

alex you want to start the game? Y/N
y
1 | 2 | 3
- + - + -
4 | 5 | 6
- + - + -
7 | 8 | 9 
 
alex choose a number available on the board: 1
X | 2 | 3
- + - + -
4 | 5 | 6
- + - + -
7 | 8 | 9 
 
007 choose a number available on the board: 2
X | X | 3
- + - + -
4 | 5 | 6
- + - + -
7 | 8 | 9 
 
alex choose a number available on the board: 3
X | X | X
- + - + -
4 | 5 | 6
- + - + -
7 | 8 | 9 
 
Congratulations alex! you won.

Do you want to play again? Y/N: 
```

The X symbol is always getting assigned to each selected case regardless of the player. :pray: fix it.

### That's it :relaxed: When finished, kindly resubmit another code review request :pray: 

**Ahmed Mahfoudh**

euqueme committed:
FIX BUGS

Add  validate_name and start_game methods to the utils file to set a default name if no name is given by the players; and to choose who starts the game or new match respectively.

Also modify validate_chip method to return when a valid piece is chosen

euqueme commented:
Hello, we fixed the bug detected by @stratospherique and also fix other bugs that were noticed by my standup team (TEAM-45-THE-HELIUMS):

* To fix the bug we modified the **validate_chip** method to return when a valid piece is chosen.

* And we added the validate_name and start_game methods to the utils file to set a default name if no name is given by the players, and to choose who starts the game or a new match respectively.

Thank you in advance to the next reviewer.

euqueme committed:
NOT_A_TIE?

Fix bug when Nobody wins message appears  if somebody wins, or ending the match before the last move is played.

bolah2009 reviewed (approved):
Good job with the game instruction. :clap: 

I would recommend we add command instructions on how to install the game (e.g. `git clone <repo>`)

It's a :+1: from me.

#### :heavy_check_mark: Approved

euqueme committed:
UPDATE INSTALLATION

Add git command line instructions to clone the repo
